features/securebackup/impl/src/main/kotlin/io/element/android/features/securebackup/impl/setup/views/RecoveryKeyView.kt

```diff
@@ -0,0 +1,221 @@
+    onClick: (() -> Unit)?,
+    onChange: ((String) -> Unit)?,
+) {
+    when (state.recoveryKeyUserStory) {
+        RecoveryKeyUserStory.Setup,
+        RecoveryKeyUserStory.Change -> RecoveryKeyStaticContent(state, onClick)
+        RecoveryKeyUserStory.Enter -> RecoveryKeyFormContent(state, onChange)
+    }
+}
+
+@Composable
+private fun RecoveryKeyStaticContent(
+    state: RecoveryKeyViewState,
+    onClick: (() -> Unit)?,
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clip(RoundedCornerShape(14.dp))
+            .background(
+                color = ElementTheme.colors.bgSubtleSecondary,
+                shape = RoundedCornerShape(14.dp)
+            )
+            .clickableIfNotNull(onClick)
+            .padding(horizontal = 16.dp, vertical = 16.dp),
+        horizontalArrangement = Arrangement.spacedBy(8.dp)
+    ) {
+        if (state.formattedRecoveryKey != null) {
+            Text(
+                text = state.formattedRecoveryKey,
+                modifier = Modifier.weight(1f),
+            )
+            Icon(
+                resourceId = CommonDrawables.ic_september_copy,
+                contentDescription = stringResource(id = CommonStrings.action_copy),
+                tint = ElementTheme.colors.iconSecondary,
+            )
+        } else {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.Center,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(vertical = 11.dp)
+            ) {
+                if (state.inProgress) {
+                    CircularProgressIndicator(
+                        modifier = Modifier
+                            .progressSemantics()
+                            .padding(end = 8.dp)
+                            .size(16.dp),
+                        color = ElementTheme.colors.textPrimary,
+                        strokeWidth = 1.5.dp,
+                    )
+                }
+                Text(
+                    text = stringResource(
+                        id = when {
+                            state.inProgress -> R.string.screen_recovery_key_generating_key
+                            state.recoveryKeyUserStory == RecoveryKeyUserStory.Change -> R.string.screen_recovery_key_change_generate_key
+                            else -> R.string.screen_recovery_key_setup_generate_key
+                        }
+                    ),
+                    textAlign = TextAlign.Center,
+                    style = ElementTheme.typography.fontBodyLgMedium,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun RecoveryKeyFormContent(state: RecoveryKeyViewState, onChange: ((String) -> Unit)?) {
+    onChange ?: error("onChange should not be null")
+    val recoveryKeyVisualTransformation = remember {
+        RecoveryKeyVisualTransformation()
+    }
+    OutlinedTextField(
```

Comment:
Nit: neither hardware nor software Enter key triggers the 'continue' action.

Reply:
Yes, I can add this